src/scripts/main.js

```diff
@@ -2,10 +2,22 @@
 
 function waitFor(element, eventName) {
   // write your code here
+  return new Promise((resolve, reject) => {
+    element.addEventListener(eventName, () => {
```

Comment:
Ensure that the `element` is not null before adding an event listener. If `element` is null, `addEventListener` will throw an error.